addons/website/static/src/js/editor/snippets.options.js

```diff
@@ -1487,15 +1491,38 @@ options.registry.topMenuColor = options.Class.extend({
     /**
      * @override
      */
-    selectStyle(previewMode, widgetValue, params) {
-        this._super(...arguments);
-        const className = widgetValue ? (params.colorPrefix + widgetValue) : '';
+    async selectStyle(previewMode, widgetValue, params) {
+        await this._super(...arguments);
+        let toggleColorOption = 'header_color';
+        if (ColorpickerDialog.isCSSColor(widgetValue)) {
+            toggleColorOption = 'header_inline_color';
+            widgetValue = `background-color:${widgetValue}`;
+        } else {
+            widgetValue = params.colorPrefix + widgetValue;
+        }
         this.trigger_up('action_demand', {
             actionName: 'toggle_page_option',
-            params: [{name: 'header_color', value: className}],
+            params: [{name: toggleColorOption, value: widgetValue}],
         });
     },
 
+    //--------------------------------------------------------------------------
+    // Public
+    //--------------------------------------------------------------------------
+
+    /**
+     * Updates the color preview box value if header_overlay value is false.
+     *
+     * @override
+     */
+    updateUI: async function () {
+        await this._super(...arguments);
+        const show = await this._computeVisibility();
+        if (!show) {
+            this.$el.find('we-toggler').removeClass('active');
+        }
+    },
```

Comment:
I think this code is useless, can you explain?

Reply:
this code is for the case we discussed before. i.e. preview box color is kept after transparent color is reset.
steps to reproduce- 
   1. apply transparent color.
   2. activate toggle.
   3. uncheck the transparent option.

Reply:
Just the fact the colorpicker stays opened? It is not about a specific fix for this option then, I think we broke the behaviour at some point and that the colorpicker should just be closed when clicking outside of it (recent tasks were merged about that and probably broke the feature). Either try to fix it in a dedicated commit here or remove this code and ignore the bug and we'll try to handle it in a separate task. Thanks :+1: